src/terrain/routes/user_info.clj

```diff
@@ -34,9 +34,9 @@
    (context "/users" []
      :tags ["admin-user-info"]
      (GET "/:username/groups" []
-          :path-params [username :- UsernameParam
-                        details :- DetailsParam]
+          :path-params [username :- UsernameParam]
+          :query [params DetailsParam]
```

Comment:
You could also destructure the `details` out of `params` here:
```suggestion
          :query [{:keys [details]} DetailsParam]
```
Then the `list-groups-for-user` function signature wouldn't have to change.

Reply:
Ah neat, I haven't seen this used anywhere else (at least not yet) - I'll refactor 👍